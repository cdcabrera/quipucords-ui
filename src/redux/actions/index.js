import { credentialsActions } from './credentialsActions';
import { factsActions } from './factsActions';
import { reportsActions } from './reportsActions';
import { scansActions } from './scansActions';
import { sourcesActions } from './sourcesActions';
import { statusActions } from './statusActions';
import { userActions } from './userActions';

const reduxActions = {
  credentials: credentialsActions,
  facts: factsActions,
  reports: reportsActions,
  scans: scansActions,
  sources: sourcesActions,
  status: statusActions,
  user: userActions
};

export {
  reduxActions as default,
  reduxActions,
  credentialsActions,
  factsActions,
  reportsActions,
  scansActions,
  sourcesActions,
  statusActions,
  userActions
};
