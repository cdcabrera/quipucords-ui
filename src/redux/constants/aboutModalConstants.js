const ABOUT_MODAL_SHOW = 'ABOUT_MODAL_SHOW';
const ABOUT_MODAL_HIDE = 'ABOUT_MODAL_HIDE';

const aboutModalTypes = {
  ABOUT_MODAL_SHOW,
  ABOUT_MODAL_HIDE
};

export { aboutModalTypes as default, aboutModalTypes, ABOUT_MODAL_SHOW, ABOUT_MODAL_HIDE };
