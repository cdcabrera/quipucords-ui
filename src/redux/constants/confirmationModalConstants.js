const CONFIRMATION_MODAL_SHOW = 'CONFIRMATION_MODAL_SHOW';
const CONFIRMATION_MODAL_HIDE = 'CONFIRMATION_MODAL_HIDE';

const confirmationModalTypes = {
  CONFIRMATION_MODAL_SHOW,
  CONFIRMATION_MODAL_HIDE
};

export { confirmationModalTypes as default, confirmationModalTypes, CONFIRMATION_MODAL_SHOW, CONFIRMATION_MODAL_HIDE };
