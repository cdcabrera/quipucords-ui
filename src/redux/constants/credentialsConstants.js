const ADD_CREDENTIAL = 'ADD_CREDENTIAL';
const DELETE_CREDENTIAL = 'DELETE_CREDENTIAL';
const GET_CREDENTIALS = 'GET_CREDENTIALS';
const UPDATE_CREDENTIAL = 'UPDATE_CREDENTIAL';
const CREATE_CREDENTIAL_SHOW = 'CREATE_CREDENTIAL_SHOW';
const EDIT_CREDENTIAL_SHOW = 'EDIT_CREDENTIAL_SHOW';
const UPDATE_CREDENTIAL_HIDE = 'UPDATE_CREDENTIAL_HIDE';

const SELECT_CREDENTIAL = 'SELECT_CREDENTIAL';
const DESELECT_CREDENTIAL = 'DESELECT_CREDENTIAL';
const EXPANDED_CREDENTIAL = 'EXPANDED_CREDENTIAL';
const NOT_EXPANDED_CREDENTIAL = 'NOT_EXPANDED_CREDENTIAL';
const UPDATE_CREDENTIALS = 'UPDATE_CREDENTIALS';

const credentialsTypes = {
  ADD_CREDENTIAL,
  UPDATE_CREDENTIAL,
  DELETE_CREDENTIAL,
  GET_CREDENTIALS,
  CREATE_CREDENTIAL_SHOW,
  EDIT_CREDENTIAL_SHOW,
  UPDATE_CREDENTIAL_HIDE,
  SELECT_CREDENTIAL,
  DESELECT_CREDENTIAL,
  EXPANDED_CREDENTIAL,
  NOT_EXPANDED_CREDENTIAL,
  UPDATE_CREDENTIALS
};

export {
  credentialsTypes as default,
  credentialsTypes,
  ADD_CREDENTIAL,
  UPDATE_CREDENTIAL,
  DELETE_CREDENTIAL,
  GET_CREDENTIALS,
  CREATE_CREDENTIAL_SHOW,
  EDIT_CREDENTIAL_SHOW,
  UPDATE_CREDENTIAL_HIDE,
  SELECT_CREDENTIAL,
  DESELECT_CREDENTIAL,
  EXPANDED_CREDENTIAL,
  NOT_EXPANDED_CREDENTIAL,
  UPDATE_CREDENTIALS
};
