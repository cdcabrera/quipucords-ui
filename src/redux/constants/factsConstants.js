const ADD_FACTS = 'ADD_FACTS';

const factsTypes = {
  ADD_FACTS
};

export { factsTypes as default, factsTypes, ADD_FACTS };
