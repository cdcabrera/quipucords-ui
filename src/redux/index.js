import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { withTranslation } from 'react-i18next';
import store from './store';
import reduxActions from './actions';
import reduxReducers from './reducers';
import reduxSelectors from './selectors';
import reduxTypes from './constants';

const connectTranslate = (mapStateToProps, mapDispatchToProps) => component =>
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(withTranslation()(component));

const connectRouterTranslate = (mapStateToProps, mapDispatchToProps) => component =>
  withRouter(connectTranslate(mapStateToProps, mapDispatchToProps)(component));

const connectRouter = (mapStateToProps, mapDispatchToProps) => component =>
  withRouter(
    connect(
      mapStateToProps,
      mapDispatchToProps
    )(component)
  );

export {
  connect,
  connectRouter,
  connectRouterTranslate,
  connectTranslate,
  reduxActions,
  reduxReducers,
  reduxSelectors,
  reduxTypes,
  store
};
