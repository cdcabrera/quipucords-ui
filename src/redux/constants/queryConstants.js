import { apiTypes } from '../../constants/apiConstants';

/**
 * Inventory query types associated with only HOSTS' and INSTANCES' queries.
 */
const SET_QUERY_CREDENTIALS_TYPES = {
  [apiTypes.API_.DIRECTION]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.DIRECTION}`,
  [RHSM_API_QUERY_TYPES.DISPLAY_NAME]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.DISPLAY_NAME}`,
  [RHSM_API_QUERY_TYPES.SORT]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.SORT}`,
  [RHSM_API_QUERY_TYPES.LIMIT]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.LIMIT}`,
  [RHSM_API_QUERY_TYPES.OFFSET]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.OFFSET}`
};
