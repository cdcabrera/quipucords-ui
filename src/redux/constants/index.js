import { aboutModalTypes } from './aboutModalConstants';
import { confirmationModalTypes } from './confirmationModalConstants';
import { credentialsTypes } from './credentialsConstants';
import { factsTypes } from './factsConstants';
import { reportsTypes } from './reportsConstants';
import { scansTypes } from './scansConstants';
import { sourcesTypes } from './sourcesConstants';
import { statusTypes } from './statusConstants';
import { toastNotificationTypes } from './toasNotificationConstants';
import { userTypes } from './userConstants';
import { viewTypes } from './viewConstants';
import { viewPaginationTypes } from './viewPaginationConstants';
import { viewToolbarTypes } from './viewToolbarConstants';

const reduxTypes = {
  aboutModal: aboutModalTypes,
  confirmationModal: confirmationModalTypes,
  credentials: credentialsTypes,
  facts: factsTypes,
  reports: reportsTypes,
  scans: scansTypes,
  sources: sourcesTypes,
  status: statusTypes,
  toastNotifications: toastNotificationTypes,
  user: userTypes,
  view: viewTypes,
  viewPagination: viewPaginationTypes,
  viewToolbar: viewToolbarTypes
};

export {
  reduxTypes as default,
  reduxTypes,
  aboutModalTypes,
  confirmationModalTypes,
  credentialsTypes,
  factsTypes,
  reportsTypes,
  scansTypes,
  sourcesTypes,
  statusTypes,
  toastNotificationTypes,
  userTypes,
  viewTypes,
  viewPaginationTypes,
  viewToolbarTypes
};
