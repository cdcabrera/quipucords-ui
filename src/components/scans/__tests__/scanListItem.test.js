import React from 'react';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import { ConnectedScanListItem, ScanListItem } from '../scanListItem';
import { viewTypes } from '../../../redux/constants';

describe('ScanListItem Component', () => {
  const generateEmptyStore = (obj = {}) => configureMockStore()(obj);

  it('should render a connected component', () => {
    const store = generateEmptyStore({ viewOptions: { [viewTypes.SCANS_VIEW]: {} } });

    const props = {
      lastRefresh: 0,
      scan: {
        jobsTotal: 1,
        id: 42,
        mostRecentEndTime: '',
        mostRecentId: 2,
        mostRecentReportId: 3,
        mostRecentStatus: 'completed',
        mostRecentStartTime: '',
        mostRecentStatusMessage: 'Lorem ipsum',
        mostRecentSysFailed: 1,
        mostRecentSysScanned: 20,
        name: 'lorem',
        sourcesTotal: 1
      }
    };

    const component = shallow(
      <Provider store={store}>
        <ConnectedScanListItem {...props} />
      </Provider>
    );

    expect(component.find(ConnectedScanListItem)).toMatchSnapshot('connected');
  });

  it('should render a non-connected component', () => {
    const props = {
      lastRefresh: 0,
      scan: {
        jobsTotal: 1,
        id: 42,
        mostRecentEndTime: '',
        mostRecentId: 2,
        mostRecentReportId: 3,
        mostRecentStatus: 'completed',
        mostRecentStartTime: '',
        mostRecentStatusMessage: 'Lorem ipsum',
        mostRecentSysFailed: 1,
        mostRecentSysScanned: 20,
        name: 'lorem',
        sourcesTotal: 1
      }
    };

    const component = shallow(<ScanListItem {...props} />);
    expect(component).toMatchSnapshot('non-connected');
  });
});
