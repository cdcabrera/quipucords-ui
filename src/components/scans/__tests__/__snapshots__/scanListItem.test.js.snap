// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SourceListItem Component should render a connected component: connected 1`] = `
<ScanListItem
  cancelScan={[Function]}
  lastRefresh={0}
  pauseScan={[Function]}
  restartScan={[Function]}
  scan={
    Object {
      "id": 42,
      "jobsTotal": 0,
      "mostRecentSysFailed": 0,
      "mostRecentSysScanned": 0,
      "name": "lorem",
      "sourcesTotal": 0,
    }
  }
  selectedScans={Array []}
  startScan={[Function]}
/>
`;

exports[`SourceListItem Component should render a non-connected component: non-connected 1`] = `
<ListViewItem
  actions={
    <Tooltip
      delayShow={500}
      id={null}
      placement="top"
      popover={null}
      rootClose={true}
      tooltip="Run Scan"
      trigger={
        Array [
          "hover",
        ]
      }
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="link"
        disabled={false}
        onClick={[Function]}
      >
        <Icon
          aria-label="Start"
          name="spinner2"
          type="pf"
        />
      </Button>
      <Connect(ScanDownload)
        downloadId={3}
        pullRight={true}
        title="Download"
      />
    </Tooltip>
  }
  additionalInfo={
    Array [
      <ListStatusItem
        count={20}
        emptyText="0 Successful"
        expandType="systemsScanned"
        expanded={false}
        iconInfo={
          Object {
            "classNames": Array [],
            "name": "ok",
            "type": "pf",
          }
        }
        id="successHosts"
        tipPlural="Successful Systems"
        tipSingular="Successful System"
        toggleExpand={[Function]}
      />,
      <ListStatusItem
        count={1}
        emptyText="0 Failed"
        expandType="systemsFailed"
        expanded={false}
        iconInfo={
          Object {
            "classNames": Array [],
            "name": "error-circle-o",
            "type": "pf",
          }
        }
        id="systemsFailed"
        tipPlural="Failed Systems"
        tipSingular="Failed System"
        toggleExpand={[Function]}
      />,
      <ListStatusItem
        count={1}
        emptyText="0 Sources"
        expandType="sources"
        expanded={false}
        iconInfo={null}
        id="sources"
        tipPlural="Sources"
        tipSingular="Source"
        toggleExpand={[Function]}
      />,
      <ListStatusItem
        count={0}
        emptyText="0 Previous"
        expandType="jobs"
        expanded={false}
        iconInfo={null}
        id="scans"
        tipPlural="Previous"
        tipSingular="Previous"
        toggleExpand={[Function]}
      />,
    ]
  }
  checkboxInput={
    <Checkbox
      bsClass=""
      checked={false}
      disabled={false}
      inline={false}
      onChange={[Function]}
      title=""
    />
  }
  className="quipucords-scan-list-item list-view-pf-top-align"
  compoundExpand={true}
  compoundExpanded={false}
  description={
    <div
      className="scan-description"
    >
      <Icon
        className="scan-status-icon"
        name="ok"
        type="pf"
      />
      <div
        className="scan-status-text"
      >
        <div>
          Lorem ipsum
        </div>
        <div
          className="text-muted"
        >
          
        </div>
      </div>
    </div>
  }
  heading={null}
  hideCloseIcon={false}
  initExpanded={false}
  key="42"
  leftContent={
    <div
      className="list-item-name"
    >
      lorem
    </div>
  }
  onCloseCompoundExpand={[Function]}
  onExpand={[Function]}
  onExpandClose={[Function]}
  stacked={false}
/>
`;
